src/main/java/racingcar/domain/Cars.java

```diff
@@ -23,29 +30,38 @@ public void addCar(Car car) {
     }
 
     public void moveAllCar() {
+        Strategy numberStrategy = new NumberStrategy();
         for (Car car : cars) {
-            car.stopOrMove(getRandomValue());
+            car.stopOrMove(numberStrategy);
         }
     }
 
-    private int getRandomValue() {
-        Random random = new Random();
-        return random.nextInt(10);
-    }
-
-    public List<Integer> getAllMoveCount() {
-        List<Integer> moveCounts = new ArrayList<>();
+    public final List<Car> getAllCar() {
+        List<Car> copyCars = new ArrayList<>();
         for (Car car : cars) {
-            moveCounts.add(car.getMoveCount());
+            copyCars.add(car.clone());
         }
-        return moveCounts;
-    }
-
-    public List<Car> getAllCar() {
-        return cars;
+        return copyCars;
     }
 
     public int getNumberOfCar() {
         return cars.size();
     }
+
+    public List<String> findWinners() {
+        List<String> winners = new ArrayList<>();
+        cars.sort(Comparator.comparingInt(Car::getMoveCount).reversed());
+        int winnerMoveCount = cars.get(0).getMoveCount();
+        decisionWinner(winners, winnerMoveCount);
+        return winners;
+    }
+
+    private void decisionWinner(List<String> winners, int winnerMoveCount) {
+        for (int i = 0; i < cars.size(); i++) {
+            Car car = cars.get(i);
+            if (car.getMoveCount() == winnerMoveCount) {
+                winners.add(car.getName());
+            }
+        }
```

Comment:
```suggestion
        for (Car car : cars) {
            if (car.getMoveCount() == winnerMoveCount) {
                winners.add(car.getName());
            }
        }
```
리스트를 직접 순회하는 방식보다는 for each 문법을 사용하는 것이 좋겠네요